pkg/util/kubernetes/apiserver/metadata_controller.go

```diff
@@ -322,9 +322,22 @@ func GetPodMetadataNames(nodeName, ns, podName string) ([]string, error) {
 		return nil, nil
 	}
 	log.Tracef("CacheKey: %s, with %d services", cacheKey, len(serviceList))
-	for _, s := range serviceList {
-		metaList = append(metaList, fmt.Sprintf("kube_service:%s", s))
-	}
-
 	return metaList, nil
 }
+
+// GetNodeLabels retrieves the labels of the queried node from the cache of the shared informer.
+func GetNodeLabels(nodeName string) (map[string]string, error) {
+	as, err := GetAPIClient()
+	if err != nil {
+		return nil, err
+	}
+	// TODO handle the wait on sync to avoid cache miss
```

Comment:
Maybe we could have `StartControllers` make sure that a nodes informer is always started since its needed for this feature? Currently, if may not be because of the config options.

`StartControllers` could also do a `WaitForCacheSync` for nodes.

Reply:
Added a wait for sync when we start the controllers